proposals/analytics.md

```diff
@@ -0,0 +1,58 @@
+# Analytics
+
+
+## Problem
+Breaking our monolithic application into several different applications has many benefits. It also introduces many levels of complexity.
+It becomes very difficult to manage complex dependencies like analytics configuration.
+This is exacerbated in the library context where, configuration, for example, must come from the application context.
+
+## Solution
+We need a cross library/application solution that we can inject configuration into at the application level.
+This solution must not leak into the component structure such that library dependencies become unusable without application configuration.
+
+### Initialisation
+In order to configure analytics we must build a library API that can be injected into:
+
+```js
+// module: utils/analytics
+
+import { init } from '@festicket/analytics';
+
+export const { track, identify }  = init({
+  ACCOUNT_KEY: process.env.ACCOUNT_KEY,
+  ACCOUNT_SECRET: process.env.ACCOUNT_SECRET,
+});
+```
+
+This means that, once configured, an application can identify or track with ease:
+
+```js
+// any file within the application
+import { track } from '@utils/analytics';
+
+// Do some stuff
+track({ type: 'MY_CUSTOM_EVENT', ...data });
+```
+
+### Automatic event capture
+This library should bind an event handle to the `document.body` so that all anchors can be tracked automatically.
+The developer must only provide `data-analytics-*` properties and `data-analytics=true` to enable the automatic tracking.
+
+```html
+<a
+  href="http://some-link.com"
+  data-analytics=true
+  data-analytics-value=test
```

Comment:
how are these values decided?

Reply:
Depends on what you're tracking.
 
For instance in some case we have the "footer_link" so people that are reading the data knows that that click comes from the footer.

Reply:
@Joebourne its up to the implementor to understand what the business needs and add that data onto the element. Hope this makes sense?

Reply:
Yep. Just wondering if there were conventions in case there were, say, two `footer_link`s to the same url.

Reply:
Segment keep a track of every event type that has been tracked, and someone from the business should be in charge of events tracking, I think that it's domain of the internal team.